build.sbt

```diff
@@ -394,6 +394,14 @@ lazy val protobufSettings = Def.settings(
   )
 )
 
+def splitTests(tests: Seq[TestDefinition], filter: Seq[String]) = {
```

Comment:
Tests run within the same group which might use or not the same Classloader. One way to guarantee diff CL is to split Tests into a separate group.

Reply:
We're assuming that users will never run into conflicting `PipelineOptions` in the same build & having to do this right?

Although I really don't see the appeal of `trait Options extends PipelineOptions` with getters/setters over case classes. If we want to really opinionated, case app should be the way to go?

OTOH there's `ScioContext#apply(opts: PipelineOptions)`, can users just do that instead of `ContextAndArgs` instead?

Reply:
yeah they will need to do smth similar to this depending on their build. Note that this change does not introduce this issue, it merely surfaced it while adding more tests.

This change is a nice to have. IMO it makes the API consistent when it comes to argument parsing. Most of our examples use `ContextAndArgs` and the ones that use `case-app` use the `ContextAndArgs.typed` variant. If users want to use `PipelineOptions` they can use what you mentioned 1) `ScioContext#apply(opts: PipelineOptions)` but can also use:
2 ) 
```
val (opts, remainingArgs) = ScioContext.parseArguments[Options](args, withValidation)
val sc = ScioContext(opts)
```

The usage is completely different and there's no reason to be. on 1) and 2) users have also the extra step of doing `sc.as(Options.class).getProperty` as opposed to direct access:

```
val (sc, {Args | CaseApp}) = ContextAndArgs
```

With this change we add:

```
val (sc, {Args | CaseApp | T <: PipelineOptions}) = ContextAndArgs
```